### ISSUE TITLE
Trackers receive the gradient from the previous iteration

### ISSUE THREAD

**Issue by cossu555** [pralab/secml-torch#207]
Description

In ModularEvasionAttackFixedEps._run_loop(), grad_before_processing is assigned after the trackers are called.

As a result, trackers receive a zero gradient during the first iteration and the gradient from the previous iteration during the following ones, , while the gradient computed during the final iteration is never recorded.

Possible fix:

Capture the current gradient immediately after _loss_and_grad() and before calling the trackers:

grad_before_processing = delta.grad.detach().clone()

This ensures that trackers receive the raw gradient from the current iteration, before gradient processing and optimizer update
